### PR TITLE
store/tikv: update conn pool.

### PR DESCRIPTION
**Pull request by disksing**
Switch to `github.com/ngaut/pools`.
@qiuyesuifeng @siddontang @shenli @coocood

### CODE REVIEW

File: store/tikv/conn_pool.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	"sync"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/ngaut/pools"
+)
+
+const poolIdleTimeoutSeconds = 120
+
+type cerateConnFunc func(addr string) (*Conn, error)
```
Comment (coocood) [inline store/tikv/conn_pool.go]: create

**Comment by coocood** [comment]
LGTM

File: store/tikv/conn_pool.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+// Close closes the pool.
+func (p *Pool) Close() {
+	p.p.Close()
+}
+
+// Pools maintains connections with multiple addrs.
+type Pools struct {
+	m sync.Mutex
+
+	capability int
```
Comment (shenli) [inline store/tikv/conn_pool.go]: This is the capacity for a single pool? Or for all the pool?

Comment (disksing) [inline store/tikv/conn_pool.go]: @shenli It's for a single pool.

**Comment by shenli** [comment]
LGTM

**Pull request merged**